uxarray/remap/nearest_neighbor.py

```diff
@@ -0,0 +1,200 @@
+        )
+
+    else:
+        raise ValueError(
+            f"Invalid coord_type. Expected either 'lonlat' or 'Cartesian', but received {coord_type}"
+        )
+
+
+def _nearest_neighbor_uxda(source_uxda: UxDataArray,
+                           destination_obj: Union[Grid, UxDataArray, UxDataset],
+                           destination_data_mapping: str = "nodes",
+                           coord_type: str = "lonlat"):
+    """Nearest Neighbor Remapping implementation for ``UxDataArray``.
+
+    Parameters
+    ---------
+    source_uxda : UxDataArray
+        Source UxDataArray for remapping
+    destination_obj : Grid, UxDataArray, UxDataset
+        Destination for remapping
+    destination_data_mapping : str, default="nodes"
+        Location of where to map data, either "nodes" or "face centers"
+    coord_type : str, default="lonlat"
+        Indicates whether to remap using on latlon or Cartesian coordinates for nearest neighbor computations when
+        remapping.
+    """
+
+    # prepare dimensions
+    if destination_data_mapping == "nodes":
+        destination_dim = "n_node"
+    else:
+        destination_dim = "n_face"
+
+    destination_dims = list(source_uxda.dims)
+    destination_dims[-1] = destination_dim
+
+    if isinstance(destination_obj, Grid):
+        destination_grid = destination_obj
+    elif isinstance(
+            destination_obj,
+        (uxarray.core.dataarray.UxDataArray, uxarray.core.dataset.UxDataset)):
+        destination_grid = destination_obj.uxgrid
+    else:
+        raise ValueError("TODO: Invalid Input")
+
+    # perform remapping
+    destination_data = nearest_neighbor(source_uxda.uxgrid, destination_grid,
+                                        source_uxda.data,
+                                        destination_data_mapping, coord_type)
+    # construct data array for remapping variable
+    uxda_remap = uxarray.core.dataarray.UxDataArray(data=destination_data,
+                                                    name=source_uxda.name,
+                                                    dims=destination_dims,
+                                                    uxgrid=destination_grid)
+    # add remapped variable to existing UxDataset
+    if isinstance(destination_obj, uxarray.core.dataset.UxDataset):
+        destination_obj[source_uxda.name] = uxda_remap
+        return destination_obj
+
+    # construct a UxDataset from remapped variable and existing variable
+    elif isinstance(destination_obj, uxarray.core.dataset.UxDataArray):
+        uxds = destination_obj.to_dataset()
+        uxds[source_uxda.name] = uxda_remap
+        return uxds
+
+    # return UxDataArray with remapped variable
+    else:
+        return uxda_remap
+
+
+def _nearest_neighbor_uxds(source_uxds: UxDataset,
+                           destination_obj: Union[Grid, UxDataArray, UxDataset],
+                           destination_data_mapping: str = "nodes",
```

Comment:
```suggestion
                           data_mapping: str = "nodes",
```

Reply:
How about changing it to something like:

`remap_to` instead of `destination_data_mapping`, since I don't want to lose the fact that we are remapping "to" something, and the data mapping doesnt reffer to the original data variable mapping.

Reply:
`remap_to` sounds very good!